lib/rubygems/commands/sources_command.rb

```diff
@@ -71,7 +75,7 @@ def check_typo_squatting(source)
 Do you want to add this source?
       QUESTION
 
-      terminate_interaction 1 unless ask_yes_no question
+      terminate_interaction 1 unless options[:force] || ask_yes_no(question)
```

Comment:
`ask_yes_no` already supports a default parameter, so it should work if you pass `options[:force]` as the second argument?

Reply:
@deivid-rodriguez yes, if you take a look at the implementation this default parameter does something different. It is only used when tty? is false and to change that contract is beyond this. I had considered an additional force option on that method to keep the existing behaviours but settled on this for safety.

Reply:
You're right :+1:.